src/service/resources-impl.js

```diff
@@ -2052,6 +2091,19 @@ export class TaskQueue_ {
   forEach(callback) {
     this.tasks_.forEach(callback);
   }
+
+  /**
+   * Dequeue tasks in queue removing any task that pass the callback test.
+   * @param {function(!TaskDef)} callback Return true to dequeue the task.
+   */
+  purge(callback) {
+    let index = this.tasks_.length;
+    while (index--) {
+      if (callback(this.tasks_[index])) {
+        this.dequeue(this.tasks_[index]);
```

Comment:
Yeah, let's get rid of the second loop since we already have all indexes we need. A single loop is fine though given that removals are rare.